Modules/TRD/src/TrackletCountCheck.cxx

```diff
@@ -0,0 +1,190 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   TrackletCountCheck.cxx
+/// \author My Name
+///
+
+#include "TRD/TrackletCountCheck.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Quality.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "Common/Utils.h"
+// ROOT
+#include <TH1.h>
+#include <TPaveText.h>
+
+#include <DataFormatsQualityControl/FlagReasons.h>
+
+using namespace std;
+using namespace o2::quality_control;
+using namespace o2::quality_control_modules::common;
+
+namespace o2::quality_control_modules::trd
+{
+
+void TrackletCountCheck::configure()
+{
+  ILOG(Debug, Devel) << "initializing TrackletCountCheck" << ENDM;
+
+  mThresholdMeanHighPerTimeFrame = getFromConfig<float>(mCustomParameters, "UpperthresholdPerTimeFrame", 520.f);
+  ILOG(Debug, Support) << "using Upperthreshold Per Timeframe= " << mThresholdMeanHighPerTimeFrame << ENDM;
+
+  mThresholdMeanLowPerTimeFrame = getFromConfig<float>(mCustomParameters, "LowerthresholdPerTimeFrame", 600.f);
+  ILOG(Debug, Support) << "using Lowerthreshold Per Timeframe= " << mThresholdMeanLowPerTimeFrame << ENDM;
+
+  mThresholdMeanHighPerTrigger = getFromConfig<float>(mCustomParameters, "UpperthresholdPerTrigger", 520.f);
+  ILOG(Debug, Support) << "using Upperthreshold Per Trigger= " << mThresholdMeanHighPerTrigger << ENDM;
+
+  mThresholdMeanLowPerTrigger = getFromConfig<float>(mCustomParameters, "LowerthresholdPerTrigger", 500.f);
+  ILOG(Debug, Support) << "using Lowerthreshold Per Trigger= " << mThresholdMeanLowPerTrigger << ENDM;
+
+  mStatThresholdPerTrigger = getFromConfig<int>(mCustomParameters, "StatThresholdPerTrigger", 1000);
+  ILOG(Debug, Support) << "using StatThreshold Per Trigger= " << mStatThresholdPerTrigger << ENDM;
+}
+
+Quality TrackletCountCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  mResultPertrigger = Quality::Null;
+  mResultPerTimeFrame = Quality::Null;
+
+  for (auto& [moName, mo] : *moMap) {
+
+    (void)moName;
+    if (mo->getName() == "trackletsperevent") {
+      auto* h = dynamic_cast<TH1F*>(mo->getObject());
+      if (h == nullptr) {
+        // ILOG(Debug, Support) << "Requested Histogram type does not match with the Histogram in source" << ENDM;
+        continue;
+      }
+      // warning about statistics available
+      TPaveText* msg1 = new TPaveText(0.3, 0.7, 0.7, 0.9, "NDC"); // check option "br","NDC"
```

Comment:
isn't this leaking memory?

Reply:
I don't know how to check memory leaks. If you help me how to check it would be good. I saw this way of writing text over hist in some other modules' class.

Reply:
well here one does not need to check, you are creating an object with `new` every time the `check` function is called and never deleting it. You could simply have a member variable plain pointer which you initialize once with a new object and that's it. Or you use a smart pointer.